specification/dnc/resource-manager/Microsoft.DelegatedNetwork/preview/2023-06-27-preview/delegatedSubnets.json

```diff
@@ -0,0 +1,471 @@
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ResourceUpdateParameters": {
+      "description": "Parameters for updating a resource.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      }
+    },
+    "DelegatedSubnets": {
+      "description": "An array of DelegatedSubnet resources.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DelegatedSubnet"
+          },
+          "description": "An array of DelegatedSubnet resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of DelegatedSubnet resources.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "DelegatedSubnet": {
+      "description": "Represents an instance of a orchestrator.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DelegatedSubnetResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DelegatedSubnetProperties",
+          "description": "Properties of the provision operation request."
+        }
+      }
+    },
+    "DelegatedSubnetProperties": {
+      "description": "Properties of delegated subnet",
+      "type": "object",
+      "properties": {
+        "resourceGuid": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource guid."
+        },
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Provisioning"
+          ],
+          "x-ms-enum": {
+            "name": "DelegatedSubnetState",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "description": "The current state of dnc delegated subnet resource."
+        },
+        "subnetDetails": {
+          "description": "subnet details",
+          "$ref": "#/definitions/subnetDetails"
+        },
+        "controllerDetails": {
+          "$ref": "./common-types.json#/definitions/controllerDetails",
+          "description": "Properties of the controller."
+        },
+        "allocationBlockPrefixSize": {
+          "description": "Defines size of CIDR blocks allocated to nodes in VnetBlock Mode.",
+          "type": "integer",
+          "format": "int32",
+          "maximum": 32,
+          "minimum": 24,
+          "default": 32
```

Comment:
Why not keeping this optional? What's the benefit of setting this default?
That way we can avoid to deal with this special value and handling it the same as in the existing subnets already stored in the db that don't have any value set for that property.